crates/pathfinder/src/state/block_hash.rs

```diff
@@ -399,33 +405,15 @@ fn compute_final_hash_pre_0_13_2(header: &BlockHeaderData) -> BlockHash {
     BlockHash(chain.finalize())
 }
 
-pub fn compute_final_hash(header: &BlockHeaderData) -> Result<BlockHash> {
-    // Concatenate the transaction count, event count, state diff length, and L1
-    // data availability mode into a single felt.
-    let mut concat_counts = [0u8; 32];
-    let mut writer = concat_counts.as_mut_slice();
-    writer
-        .write_all(&header.transaction_count.to_be_bytes())
-        .unwrap();
-    writer.write_all(&header.event_count.to_be_bytes()).unwrap();
-    writer
-        .write_all(&header.state_diff_length.to_be_bytes())
-        .unwrap();
-    writer
-        .write_all(&[match header.l1_da_mode {
-            L1DataAvailabilityMode::Calldata => 0,
-            L1DataAvailabilityMode::Blob => 0b10000000,
-        }])
-        .unwrap();
-    let concat_counts = MontFelt::from_be_bytes(concat_counts);
+fn compute_final_hash_old(header: &BlockHeaderData) -> Result<BlockHash> {
```

Comment:
I would maybe refer to these names with their versions (as opposed to _"old"_ or _"new"_).

So maybe `compute_hash_v0` or `compute_v0_hash` (or anything along those lines).